plugins/vpp/ipsecplugin/vppcalls/dump_vppcalls.go

```diff
@@ -0,0 +1,337 @@
+}
+
+// IPSecSaMeta contains all VPP-specific metadata
+type IPSecSaMeta struct {
+	SaID           uint32
+	Interface      string
+	IfIdx          uint32
+	CryptoKeyLen   uint8
+	IntegKeyLen    uint8
+	Salt           uint32
+	SeqOutbound    uint64
+	LastSeqInbound uint64
+	ReplayWindow   uint64
+	TotalDataSize  uint64
+}
+
+func (handler *ipSecVppHandler) DumpIPSecSA() (saList []*IPSecSaDetails, err error) {
+	return handler.DumpIPSecSAWithIndex(^uint32(0)) // Get everything
+}
+
+func (handler *ipSecVppHandler) DumpIPSecSAWithIndex(saID uint32) (saList []*IPSecSaDetails, err error) {
+	defer func(t time.Time) {
+		handler.stopwatch.TimeLog(ipsecapi.IpsecSaDump{}).LogTimeEntry(time.Since(t))
+	}(time.Now())
+
+	saDetails, err := handler.dumpSecurityAssociations(saID)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, saData := range saDetails {
+		// Skip tunnel interfaces
+		if saData.SwIfIndex != ^uint32(0) {
+			continue
+		}
+
+		// Addresses
+		var tunnelSrcAddrStr, tunnelDstAddrStr string
+		if uintToBool(saData.IsTunnelIP6) {
+			var tunnelSrcAddr, tunnelDstAddr net.IP = saData.TunnelSrcAddr, saData.TunnelDstAddr
+			tunnelSrcAddrStr, tunnelDstAddrStr = tunnelSrcAddr.String(), tunnelDstAddr.String()
+		} else {
+			var tunnelSrcAddr, tunnelDstAddr net.IP = saData.TunnelSrcAddr[:4], saData.TunnelDstAddr[:4]
+			tunnelSrcAddrStr, tunnelDstAddrStr = tunnelSrcAddr.String(), tunnelDstAddr.String()
+		}
+
+		sa := &ipsec.SecurityAssociations_SA{
+			Spi:            saData.Spi,
+			Protocol:       ipsec.SecurityAssociations_SA_IPSecProtocol(saData.Protocol),
+			CryptoAlg:      ipsec.CryptoAlgorithm(saData.CryptoAlg),
+			CryptoKey:      string(saData.CryptoKey[:saData.CryptoKeyLen]),
+			IntegAlg:       ipsec.IntegAlgorithm(saData.IntegAlg),
+			IntegKey:       string(saData.IntegKey[:saData.IntegKeyLen]),
+			UseEsn:         uintToBool(saData.UseEsn),
+			UseAntiReplay:  uintToBool(saData.UseAntiReplay),
+			TunnelSrcAddr:  tunnelSrcAddrStr,
+			TunnelDstAddr:  tunnelDstAddrStr,
+			EnableUdpEncap: uintToBool(saData.UDPEncap),
+		}
+		meta := &IPSecSaMeta{
+			SaID:           saData.SaID,
+			IfIdx:          saData.SwIfIndex,
+			CryptoKeyLen:   saData.CryptoKeyLen,
+			IntegKeyLen:    saData.IntegKeyLen,
+			Salt:           saData.Salt,
+			SeqOutbound:    saData.SeqOutbound,
+			LastSeqInbound: saData.LastSeqInbound,
+			ReplayWindow:   saData.ReplayWindow,
+			TotalDataSize:  saData.TotalDataSize,
+		}
+		saList = append(saList, &IPSecSaDetails{
+			Sa:   sa,
+			Meta: meta,
+		})
+	}
+
+	return saList, nil
+}
+
+// IPSecTunnelInterfaceDetails hold a list of tunnel interfaces with name/index map as metadata
+type IPSecTunnelInterfaceDetails struct {
+	Tunnels *ipsec.TunnelInterfaces
+	Meta    *IPSecTunnelMeta
+}
+
+// IPSecTunnelMeta contains map of name/index pairs
+type IPSecTunnelMeta struct {
+	IfNameToIdx map[uint32]string
+}
+
+func (handler *ipSecVppHandler) DumpIPSecTunnelInterfaces() (tun *IPSecTunnelInterfaceDetails, err error) {
```

Comment:
I don't understand why this is different from the other dumps. Other dumps return **slice** or **map** of `xxxDetails` structs. This one returns single `IPSecTunnelInterfaceDetails` struct with all the tunnels inside. 

Why not use the same way everywhere?